cpp/src/parquet/column_reader.cc

```diff
@@ -374,14 +381,10 @@ std::shared_ptr<Page> SerializedPageReader::NextPage() {
 
       page_buffer = decryption_buffer_;
     }
-    // Uncompress it if we need to
-    if (decompressor_ != nullptr) {
-      page_buffer = DecompressPage(compressed_len, uncompressed_len, page_buffer->data());
-    }
 
-    const PageType::type page_type = LoadEnumSafe(&current_page_header_.type);
+    current_page_type_ = LoadEnumSafe(&current_page_header_.type);
```

Comment:
sorry, its not immediately obvious why thi sis becoming a member?

Reply:
Hmm, this was needed in a previous version of this patch, but I'll revert.